### PR TITLE
[bikeshed] WIP new w3c html

### PR DESCRIPTION
**Pull request by ngzhian**
This uses a multihtml Sphinx builder, together with custom HTML layout
(layout.html and page.html), to build a document that matches w3c
requirements.

Currently the toctree directive does not work right, so it is duplicated
in the side bar (https://github.com/sphinx-doc/sphinx/pull/9830). The
W3C styles also don't apply fully to the side bar (because the Sphinx
generated TOC doesn't have the right class name, but this might be fixed
by some JS).

layout.html is used for all pages, index + individual pages, so the
front matter is guarded by a check that the current page is index.
layout.html is rather large, most of it is the front matter copied from
https://github.com/WebAssembly/spec/issues/1386#issuecomment-957347678.
page.html is used by all the individual rst pages.

Right now all the configuration lives in a new folder, w3chtml, but it
can presumably replace the w3chtml folder.

### CODE REVIEW

**Comment by ericprud** [comment]
Many thanks for the W3C doc work.

Just as an FYI, W3C specs don't have to be monolithic; it's up to us to publish as a single doc (maybe nice for folks who want to print?) or a multiple pages (more responsive). It might be possible to publish both, but I imagine that simply publishing a multi-page doc with a link to a printable version would be simpler.

Also, we don't have to work all this out before publishing.

File: document/core/index.rst [hunk 1]
```diff
@@ -10,16 +10,19 @@ WebAssembly Specification
...
-   intro/index
-   syntax/index
-   valid/index
-   exec/index
-   binary/index
-   text/index
-   appendix/index
+.. only:: builder_html
+
+  .. toctree::
```
Comment (rossberg) [inline document/core/index.rst:15]: Won't this remove the toctree from the PDF?

File: document/core/Makefile [hunk 2]
```diff
@@ -120,6 +120,22 @@ html: index
 	@echo
 	@echo "Build finished. The HTML pages are in `pwd`/$(BUILDDIR)/html/."
 
+.PHONY: newhtml
```
Comment (rossberg) [inline document/core/Makefile:123]: Should we call this `w3chtml` perhaps?

File: document/core/conf.py [hunk 3]
```diff
@@ -62,8 +62,12 @@
 title = u'WebAssembly Specification'
 copyright = u'2017, WebAssembly Community Group'
 author = u'WebAssembly Community Group'
-editor = u'Andreas Rossberg (editor)'
+editor_name = u'Andreas Rossberg'
+editor = editor_name + u' (editor)'
```
Comment (rossberg) [inline document/core/conf.py:66]: Nit: rename editor_name to editor and inline the "(editor)" where needed? (I see only one occurrence anyway)

**Comment by ngzhian** [comment]
> Many thanks for the W3C doc work.
> 
> Just as an FYI, W3C specs don't have to be monolithic; it's up to us to publish as a single doc (maybe nice for folks who want to print?) or a multiple pages (more responsive). It might be possible to publish both, but I imagine that simply publishing a multi-page doc with a link to a printable version would be simpler.
> 
> Also, we don't have to work all this out before publishing now; we can add features later.

the PR title is a misnomer, it should just be "new html". I think I called it new single html because the build step followed the singlehtml. Let me fix that.

**Comment by rossberg** [comment]
@ngzhian, what's the status of this PR, is it still needed?

**Comment by ngzhian** [comment]
Not actively working on this, you can close it if you'ld like.

**Pull request closed without merging**